src/rantanplan/core.py

```diff
@@ -491,4 +649,86 @@ def get_scansion(text, rhyme_analysis=False, rhythm_format="pattern"):
                     else:
                         line["rhyme_type"] = rhyme["rhyme_type"]
                         line["rhyme_relaxation"] = rhyme["rhyme_relaxation"]
+        for idx, line in enumerate(lines):
+            structure = STRUCTURES_LENGTH.get(line["structure"], None)
+            if structure is not None:
+                if line["rhythm"]["length"] < structure[idx]:
+                    pass
+    """
+    # This is for testing with Navarro Colorado's sonnet corpus only
+    for idx, line in enumerate(lines):
+        if line["rhythm"]["length"] < 11:
+            for candidate in generate_phonological_groups(raw_tokens[idx]):
+                rhythm = get_rhythmical_pattern(candidate, rhythm_format)
+                if rhythm["length"] == 11 and rhythm["stress"][-1] == "-":
+                    line.update({
+                       "phonological_groups": candidate,
+                       "rhythm": rhythm,
+                    })
+                    break
+    """
     return lines
+
+
+def generate_phonological_groups(tokens):
+    """
+    Generates phonological groups from a list of tokens
+    :param tokens: list of spaCy tokens
+    :return: Generator with a list of phonological groups
+    """
+    for alternative_syllabification in (True, False):
+        words = get_words(tokens, alternative_syllabification)
+        syllables = get_syllables_word_end(words)
+        for liaison in (
+                ("synalepha",),
+                ("sinaeresis",),
+                ("sinaeresis", "synalepha"),
+                ("synalepha", "sinaeresis"),
+        ):
+            break_on_h = lambda liaison_type, syll1, syll2: (
+                    liaison_type == "synalepha"
+                    and syll2["syllable"][0].lower() == "h"
+            )
```

Comment:
Let's change it to:
```            def break_on_h(liaison_type, syllable_left, syllable_right):
                return (
                    liaison_type == "synalepha"
                    and syllable_right["syllable"][0].lower() == "h"
                )
```
And take it out of the for loops. We might take it to a different file if we add more breakage functions.